packages/common/core-utils/src/layerCompat.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Result of a layer compatibility check.
+ * @internal
+ */
+export type LayerCompatCheckResult =
+	| { readonly compatible: true }
+	| {
+			readonly compatible: false;
+			readonly generationCompatible: boolean;
```

Comment:
I prefer naming boolean variables with `is` question, i.e. `isGenerationCompatible`

Reply:
`isCompatible` is weird though because the subject of the sentence would be plural ("the layers"), but `areCompatible` seems very unnatural to me.

Could consider restating this type as `LayerIncompatibilityReason` and have `checkLayerCompatibility` return `undefined` if they're compatible?  Not sure about that either TBH

Reply:
Changed it again to a free function. Please take another look.